storage-node/src/services/sync/tasks.ts

```diff
@@ -138,9 +138,9 @@ export class DownloadFileTask implements SyncTask {
         .set('X-COLOSSUS-HOST-ID', this.hostId) as unknown as NodeJS.ReadableStream
       const fileStream = fs.createWriteStream(tempFilePath)
 
-      request.on('response', (res) => {
-        if (!res.ok && res.statusCode !== 404) {
-          logger.warn(`Sync - unexpected status code(${res.statusCode}) for ${res?.request?.url}`)
+      request.on('response', (res: Response) => {
+        if (!res.ok) {
+          request.emit('error', `request failed: ${res.error}`)
```

Comment:
Seeing that request is a `Stream` therefore I assume emitting error here will throw an exception in later part of the code.. at `await streamPipeline(request, fileStream)` ?

Reply:
That's correct, I also tested this specific case locally.